plugins/completion/copilot-vim.nix

```diff
@@ -12,33 +12,72 @@ with helpers.vim-plugin;
     originalName = "copilot.vim";
     defaultPackage = pkgs.vimPlugins.copilot-vim;
     globalPrefix = "copilot_";
+
+    # TODO introduced 2024-03-02: remove 2024-05-02
     deprecateExtraConfig = true;
+    optionsRenamedToSettings = [
+      "nodeCommand"
+      "filetypes"
+      "proxy"
+    ];
 
-    options = {
-      nodeCommand = mkDefaultOpt {
-        global = "node_command";
+    settingsOptions = {
+      node_command = mkOption {
         type = types.str;
```

Comment:
I know it was not the case before, but this should be a `nullOr; types.str`, maybe the user does not want to set it to use the environement provided nodejs

Reply:
Done !